bond/bond.go

```diff
@@ -102,18 +103,23 @@ func checkLinkExists(linkName string, netNsHandle *netlink.Handle) (netlink.Link
 }
 
 // configure the bonded link & add it using the netNsHandle context to add it to the required namespace. return a bondLinkObj pointer & error
-func createBondedLink(bondName string, bondMode string, bondMiimon string, failOverMac int, netNsHandle *netlink.Handle) (*netlink.Bond, error) {
+func createBondedLink(bondName string, bondMode string, bondMiimon string, bondMTU string, failOverMac int, netNsHandle *netlink.Handle) (*netlink.Bond, error) {
 	var err error
 	bondLinkObj := netlink.NewLinkBond(netlink.NewLinkAttrs())
 	bondModeObj := netlink.StringToBondMode(bondMode)
 	bondLinkObj.Attrs().Name = bondName
 	bondLinkObj.Mode = bondModeObj
 	bondLinkObj.Miimon, err = strconv.Atoi(bondMiimon)
-	bondLinkObj.FailOverMac = netlink.BondFailOverMac(failOverMac)
-
 	if err != nil {
 		return nil, fmt.Errorf("Failed to convert bondMiimon value (%+v) to an int, error: %+v", bondMiimon, err)
 	}
+	if len(bondMTU) > 0 {
```

Comment:
do we also want to validate that the VFs we are going to attach have the same/higher MTU before we do it?

Reply:
The VF mtu's will be automatically changed to that of the bond.